anndata/core/anndata.py

```diff
@@ -1439,10 +1428,13 @@ def copy(self, filename: Optional[PathLike] = None) -> 'AnnData':
                 varm=self.varm.copy(),
                 obsp=self.obsp.copy(),
                 varp=self.varp.copy(),
-                raw=None if self._raw is None else self._raw.copy(),
                 layers=self.layers.copy(),
                 dtype=dtype,
             )
+            if self._raw is not None:
+                adata._raw = self._raw.copy()
```

Comment:
Out of scope. There hasn’t been any before. Let’s do this after this PR

Reply:
?

It was passed through the AnnData constructor before, I think it should still do that. There is validation that happens there.

Reply:
Ah, whoops, that’s the `copy` one not the `__init__` one.

That makes validation completely useless, since `adata` is a copy of `self`, so everything should be the same.

Reply:
Should be the same, but we also should check. Better this throws an error than doesn't.